### ISSUE TITLE
Heroku Timeouts Not Being Monitored in Scout APM

### ISSUE THREAD

**Issue by berniechiu** [scoutapp/scout_apm_ruby#499]
### Application
Rails: `7.0.8`
Rack: `2.2.9`
Scout: `5.3.5`

### Description

Hi Scout Team,

We're experiencing [Heroku timeouts events](https://devcenter.heroku.com/articles/request-timeout) occassionly, spotted on basic metrics on Heroku, but we've never seen those timeouts recorded in Scout. I think it's hard to investigate the root cause if Scout is not properly monitoring here.

### Screenshots

![Screen Shot 2024-05-07 at 10 36 06 AM](https://github.com/scoutapp/scout_apm_ruby/assets/2749593/36fef3fc-edc2-4313-a312-a0d7fc6d0906)
![Screen Shot 2024-05-07 at 10 36 47 AM](https://github.com/scoutapp/scout_apm_ruby/assets/2749593/b5e89257-7a2f-451f-81b2-8abcabe54d26)

**Comment by lancetarn** [comment]
This is true. We have not generalized the handling of these. Are you using [rack-timeout](https://github.com/zombocom/rack-timeout) by chance? That puts the signal back in a place where we can more easily deal with it and do cleanup in our instrumentation, close spans and send them before everything exits. We have some ideas around middleware for this situation. I'll see what we can share.
